internal/helpers/database/database.go

```diff
@@ -33,11 +33,17 @@ import (
 func FetchDatabaseData(ctx context.Context, dbcr *kindav1beta1.Database, dbCred database.Credentials, instance *kindav1beta1.DbInstance) (database.Database, *database.DatabaseUser, error) {
 	log := log.FromContext(ctx)
 	host := instance.Status.Info["DB_CONN"]
-	port, err := strconv.Atoi(instance.Status.Info["DB_PORT"])
+	port64, err := strconv.ParseUint(instance.Status.Info["DB_PORT"], 10, 16)
 	if err != nil {
 		log.Error(err, "can't get port information from the instanceRef")
 		return nil, nil, err
 	}
+	if port64 > 65535 {
```

Comment:
I meant, it should not be required to check whether a uint16 value is greater than 65535, because when you convert a string to a uint64 with a size 16, it's already checking the size and it will throw an error if it's greater than 65535, so you  can remove the whole **if** statement

Reply:
good